### PR TITLE
Fix the support of `arguments` in private `get/set` method

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Ref: https://github.com/babel/babel/pull/16299

In this case, the output becomes worse, but this should be rare.
We can still reduce some outputs by passing the `this` as a parameter.

### CODE REVIEW

File: packages/babel-helpers/src/helpers/curryThis.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* @minVersion 7.24.0 */
+
+export default function _curryThis(fn: Function) {
+  return function (thisArg: any) {
+    return fn.apply(thisArg, [].slice.call(arguments, 1));
+  };
+}
```
Comment (nicolo-ribaudo) [inline packages/babel-helpers/src/helpers/curryThis.ts:7]: Equivalent to
```js
export default function _curryThis(fn: Function) {
  return function (thisArg: any) {
    return fn.call(...arguments);
  }
}
```
equivalent to
```js
export default function _curryThis(fn: Function) {
  return function (thisArg: any) {
    return Function.call.bind(fn)(...arguments);
  }
}
```
equivalent to
```js
export default function _curryThis(fn: Function) {
  var bound = Function.call.bind(fn);
  return function (thisArg: any) {
    return bound(...arguments);
  }
}
```
equivalent to
```js
export default function _curryThis(fn: Function) {
  return Function.call.bind(fn);
}
```
equivalent to
```js
export default function _curryThis(fn: Function) {
  return fn.call.bind(fn);
}
```
:)

File: packages/babel-plugin-transform-private-methods/test/fixtures/accessors/arguments/exec.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class Cl {
+  #privateField = "top secret string";
+
+  constructor() {
+    this.publicField = "not secret string";
+  }
+
+  get #privateFieldValue() {
+    expect(arguments[0]).toBeUndefined();
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-private-methods/test/fixtures/accessors/arguments/exec.js:9]:
```suggestion
    expect(arguments.length).toBe(0);
```

File: packages/babel-plugin-transform-private-methods/test/fixtures/accessors/arguments/input.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Cl {
+  #privateField = "top secret string";
+
+  constructor() {
+    this.publicField = "not secret string";
+  }
+
+  get #privateFieldValue() {
+    return this.#privateField;
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-private-methods/test/fixtures/accessors/arguments/input.js:9]: Use `arguments` somewhere here maybe, to show the outputs for getters too.

**Review by nicolo-ribaudo — COMMENTED** [review]
As an alternative, could we avoid wrapping the function and instead rewrite `arguments` usage?
1. `arguments[(expr)]` -> `arguments[1 + (expr)]`
2. `arguments.length` -> `arguments.length - 1`
3. Any other usage of `arguments` -> add `var _arguments = [].slice.call(arguments, 1)` at the beginning of the fn, and replace `arguments` with `_arguments`.

**Comment by liuxingbaoyu** [comment]
I implemented the third situation because the other two seemingly rare and may affect readability.

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]

**Pull request merged**